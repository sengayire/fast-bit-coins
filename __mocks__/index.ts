jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {})
      }
    }
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {}
  }
}))

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn()
}))
