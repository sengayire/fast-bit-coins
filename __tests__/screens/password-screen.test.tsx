import { NavigationContainer } from '@react-navigation/native'
import { render, waitFor, screen } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import '../../__mocks__'
import { MockedProvider } from '@apollo/client/testing'
import PasswordScreen from 'src/screens/password-screen'

const mockState = {
  user: {
    email: 'prince@prince.prince',
    password: 'Daprince@11',
    confirmPassword: 'Daprince@11',
    country: 'United State',
    state: 'Alabama'
  }
}

describe('Password screen', () => {
  const mocks = []

  beforeEach(() => {
    jest.mock('..')
  })
  it('should render password screen', async () => {
    const mockStore = configureStore()
    const store = mockStore(mockState)

    render(
      <NavigationContainer>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Provider store={store}>
            <PasswordScreen />
          </Provider>
        </MockedProvider>
      </NavigationContainer>
    )
    await waitFor(() => expect(screen.getByTestId('password')).toHaveTextContent)
  })
})
