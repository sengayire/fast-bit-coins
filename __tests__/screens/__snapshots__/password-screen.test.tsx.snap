// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password screen should render password screen 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingTop": 0,
      "paddingVertical": 28,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          [
            {
              "color": "#707070",
            },
            {
              "fontSize": 14,
              "fontWeight": 400,
              "lineHeight": 18,
              "paddingHorizontal": 48,
              "paddingVertical": 19,
              "textAlign": "center",
            },
          ]
        }
      >
        password.text
      </Text>
    </View>
    <View
      style={
        {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <Text
        style={
          {
            "color": "#707070",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
            "paddingBottom": 8,
          }
        }
      >
        password.input.label
      </Text>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#101820",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "transform": [
              {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          onChangeText={[Function]}
          placeholder="password.input.placeholder"
          placeholderTextColor="#86939e"
          secureTextEntry={true}
          style={
            {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
              "padding": 18,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <Text
        style={
          {
            "color": "#707070",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
            "paddingBottom": 8,
          }
        }
      >
        password.confirm.input.label
      </Text>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#101820",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "transform": [
              {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          onChangeText={[Function]}
          placeholder="password.confirm.input.placeholder"
          placeholderTextColor="#86939e"
          secureTextEntry={true}
          style={
            {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
              "padding": 18,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E7E8E9",
          "borderRadius": 5,
          "color": "#CFD1D2",
          "elevation": 3,
          "flex": 1,
          "height": 111.42857142857143,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "#E7E8E9",
            "color": "#CFD1D2",
            "fontSize": 21,
            "fontWeight": "600",
            "lineHeight": 32,
          }
        }
      >
        language.select.continue.btn
      </Text>
    </View>
  </View>
</View>
`;
